packages/gatsby/lib/query-runner/query-compiler.js

```diff
@@ -0,0 +1,127 @@
+// @flow
+import path from 'path'
+import glob from 'glob'
+import Bluebird from 'bluebird'
+import invariant from 'invariant'
+import { IRTransforms } from 'relay-compiler'
+import ASTConvert from 'relay-compiler/lib/ASTConvert'
+import RelayPrinter from 'relay-compiler/lib/RelayPrinter'
+import RelayCompilerContext from 'relay-compiler/lib/RelayCompilerContext'
+import filterContextForNode from 'relay-compiler/lib/filterContextForNode'
+
+import { store } from '../redux'
+import FileParser from './file-parser'
+
+import type { DocumentNode, GraphQLSchema } from 'graphql'
+
+const { printTransforms } = IRTransforms
+const globp = Bluebird.promisify(glob)
+
+const {
+  ArgumentsOfCorrectTypeRule,
+  DefaultValuesOfCorrectTypeRule,
+  FragmentsOnCompositeTypesRule,
+  KnownTypeNamesRule,
+  LoneAnonymousOperationRule,
+  PossibleFragmentSpreadsRule,
+  ScalarLeafsRule,
+  VariablesAreInputTypesRule,
+  VariablesInAllowedPositionRule,
+} = require(`graphql`)
+
+type RootQuery = {
+  name: string,
+  path: string,
+  text: string,
+}
+
+class Runner {
+  baseDir: string
+  schema: GraphQLSchema
+
+  constructor(baseDir: string, schema: GraphQLSchema) {
+    this.baseDir = baseDir
+    this.schema = schema
+  }
+
+  async compileAll() {
+    let nodes = await this.parseEverything()
+    return this.write(nodes)
+  }
+
+  async parseEverything() {
+    let files = await globp(`${this.baseDir}/**/*.js`)
+    let parser = new FileParser()
+
+    return await parser.parseFiles(files)
+  }
+
+  async write(nodes: Map<string, DocumentNode>) {
```

Comment:
Most of this is an adaptation of the FileWriter exported from `relay-compiler` but skips all the extra codegen work, I am hoping that the API in the future will be a bit more amenable to customizing what they export to avoid some of this leaky compiler acces